src/sentry/static/sentry/app/views/routeError.jsx

```diff
@@ -36,7 +36,19 @@ class RouteError extends React.Component {
 
     const route = getRouteStringFromRoutes(routes);
     if (route) {
-      error.message = `${error.message}: ${route}`;
+      /**
+       * Unexpectedly, error.message would sometimes not have a setter property, causing another exception to be thrown,
+       * and losing the original error in the process. Wrapping the mutation in a try-catch in an attempt to preserve
+       * the original error for logging.
+       * See https://github.com/getsentry/sentry/issues/16314 for more details.
+       */
+      try {
+        error.message = `${error.message}: ${route}`;
+      } catch (e) {
+        if (!(e instanceof TypeError)) {
+          throw e;
```

Comment:
Let's capture in Sentry instead of re-throwing, otherwise we'll have a similar issue.

Reply:
Done.